hatchet/readers/hpctoolkit_reader_latest.py

```diff
@@ -0,0 +1,421 @@
+
+            file.seek(pMetrics)
+            meta_db = file.read(szMetrics)
+
+        pMetrics_old = pMetrics
+        (pMetrics, nMetrics, szMetric, szScopeInst) = safe_unpack("<QLBB", meta_db, 0)
+
+        for i in range(nMetrics):
+            (pName, pScopeInsts, _, nScopeInsts) = safe_unpack(
+                "<QQQH", meta_db, pMetrics - pMetrics_old, i, szMetric
+            )
+
+            name = read_string(meta_db, pName - pMetrics_old).lower().strip()
+            unit = None
+            if name.endswith(")"):
+                name = name[:-1]
+                unit = name.split("(")[1].lower().strip()
+                name = name.split("(")[0].lower().strip()
+
+            for j in range(nScopeInsts):
+                (pScope, propMetricId) = safe_unpack(
+                    "<QH", meta_db, pScopeInsts - pMetrics_old, j, szScopeInst
+                )
+                (pScopeName,) = safe_unpack("<Q", meta_db, pScope - pMetrics_old)
+                scope_name = METRIC_SCOPE_MAPPING[
+                    read_string(meta_db, pScopeName - pMetrics_old).lower().strip()
+                ]
+
+                if name in ["cputime", "realtime", "cycles"] and scope_name == "i":
+                    self._time_metric = f"{name} ({scope_name})"
+
+                if scope_name in ["i", "e"]:
+                    self._metric_descriptions[propMetricId] = f"{name} ({scope_name})"
+
+    def _parse_source_file(self, meta_db: bytes, pFile: int) -> Dict[str, str]:
+        if pFile not in self._source_files:
+            (pPath,) = safe_unpack(
+                "<Q",
+                meta_db,
+                pFile + struct.calcsize("<Q"),
+            )
+            self._source_files[pFile] = {
+                "id": pFile,
+                "file_path": read_string(meta_db, pPath),
+            }
+
+        return self._source_files[pFile]
+
+    def _parse_load_module(self, meta_db: bytes, pModule: int) -> Dict[str, str]:
+        if pModule not in self._load_modules:
+            (pPath,) = safe_unpack(
+                "<Q",
+                meta_db,
+                pModule + struct.calcsize("<Q"),
+            )
+            self._load_modules[pModule] = {
+                "id": pModule,
+                "module_path": read_string(meta_db, pPath),
+            }
+
+        return self._load_modules[pModule]
+
+    def _parse_function(
+        self, meta_db: bytes, pFunction: int
+    ) -> Dict[str, Union[str, int]]:
+        if pFunction not in self._functions:
+            (pName, pModule, offset, pFile, line) = safe_unpack(
+                "<QQQQL", meta_db, pFunction
+            )
+
+            name = read_string(meta_db, pName)
+
+            if re.fullmatch(
+                "P?MPI_.+",
+                name,
+            ):
+                if name.startswith("P"):
+                    name = name[1:]
+
+                name = name[: re.match("^MPI_[a-zA-Z_]+", name).end()]
```

Comment:
@pearce8 do we want all PMPI functions to be treated as equivalent to their corresponding MPI functions? If not, this `if` statement should be removed.

Reply:
@ilumsden what does "equivalent" mean in this context?

Reply:
Literally, what this code does is it renames nodes that start with `PMPI_` so that they start with `MPI_` instead. That's what I meant by "equivalent".

Reply:
@draganagrbicllnl what is the intent here?  Usually we will have MPI call in application -> PMPI hook to potentially re-direct to a tool like Caliper -> actual MPI function (then back up the stack).  What does it look like from HPC toolkit, and what do we want to show to the user?  I think nested MPI functions would be confusing to the user.

Reply:
@pearce8 fixed it